genesis/src/main.rs

```diff
@@ -523,10 +680,93 @@ mod tests {
             });
 
             assert_ne!(i, 0);
+
+            assert_eq!(
+                primordial_accounts2[&serde_json::to_string(&keypair.to_bytes().to_vec()).unwrap()]
+                    .owner,
+                genesis_block.accounts[i].1.owner.to_string(),
+            );
+
             assert_eq!(
-                primordial_accounts2[&serde_json::to_string(&keypair.to_bytes().to_vec()).unwrap()],
+                primordial_accounts2[&serde_json::to_string(&keypair.to_bytes().to_vec()).unwrap()]
+                    .balance,
                 genesis_block.accounts[i].1.lamports,
             );
+
+            assert_eq!(
+                primordial_accounts2[&serde_json::to_string(&keypair.to_bytes().to_vec()).unwrap()]
+                    .executable,
+                genesis_block.accounts[i].1.executable,
+            );
+
+            assert_eq!(
+                primordial_accounts2[&serde_json::to_string(&keypair.to_bytes().to_vec()).unwrap()]
+                    .data,
+                base64::encode(&genesis_block.accounts[i].1.data),
+            );
         });
     }
+
+    #[test]
+    fn test_primordial_account_struct_compatibility() {
+        let yaml_string_pubkey = "---
+98frSc8R8toHoS3tQ1xWSvHCvGEADRM9hAm5qmUKjSDX:
+  balance: 3
+  owner: Gw6S9CPzR8jHku1QQMdiqcmUKjC2dhJ3gzagWduA6PGw
+  data: bWUgaGVsbG8gdG8gd29ybGQ=
+  executable: true
+6s36rsNPDfRSvzwek7Ly3mQu9jUMwgqBhjePZMV6Acp4:
+  balance: 2
+  owner: DBC5d45LUHTCrq42ZmCdzc8A8ufwTaiYsL9pZY7KU6TR
+  data: aGVsbG8=
+  executable: false
+8Y98svZv5sPHhQiPqZvqA5Z5djQ8hieodscvb61RskMJ:
+  balance: 1
+  owner: DSknYr8cPucRbx2VyssZ7Yx3iiRqNGD38VqVahkUvgV1
+  data: aGVsbG8gd29ybGQ=
+  executable: true";
+
+        let path = Path::new("test_genesis_pubkey.yaml");
```

Comment:
Use https://docs.rs/tempfile/3.1.0/tempfile/fn.tempfile.html please, so the tmp files will get cleaned up automatically